llama_index/response_synthesizers/refine.py

```diff
@@ -170,12 +170,16 @@ def _give_response_single(
         for cur_text_chunk in text_chunks:
             query_satisfied = False
             if response is None and not self._streaming:
-                structured_response = cast(
-                    StructuredRefineResponse, program(context_str=cur_text_chunk)
-                )
-                query_satisfied = structured_response.query_satisfied
-                if query_satisfied:
-                    response = structured_response.answer
+                try:
+                    structured_response = cast(
+                        StructuredRefineResponse, program(context_str=cur_text_chunk)
+                    )
+                    query_satisfied = structured_response.query_satisfied
+                    if query_satisfied:
+                        response = structured_response.answer
+                except Exception as e:
+                    logger.error(f'Error: {e}', exc_info=True)
+                    return response
             elif response is None and self._streaming:
```

Comment:
Sometime the issue is occuring at first chuck itself so added it so that some answer can be retrieved from other chunks.